docker/rp/Dockerfile

```diff
@@ -0,0 +1,37 @@
+
+# RP DOCKERFILE
+
+# The ExaWorks SDK Base
+# this expectes something like `docker build base -t base`
+FROM base
+
+# Install dependencies
+RUN yum -y install openmpi python3-pip
+
+# MongoDB install
+RUN echo -e "\
+[mongodb]\n\
+name     = MongoDB Repository \n\
+baseurl  = https://repo.mongodb.org/yum/redhat/7Server/mongodb-org/3.0/x86_64/ \n\
+gpgcheck = 0 \n\
+enabled  = 1 \n" >> /etc/yum.repos.d/mongodb.repo
+
+RUN yum update -y && yum -y install mongodb-org
+RUN mkdir -p /data/db /var/log/mongodb /var/run/mongodb
+ENV RADICAL_PILOT_DBURL=mongodb://localhost/am
+
+# RP install
+ENV VIRTUAL_ENV=/ve_rp
+RUN python3 -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+RUN which python3
+RUN pip install --upgrade pip setuptools
+RUN pip install pytest
+RUN pip install radical.pilot
```

Comment:
Unfortunately with Docker, each `RUN` results in a whole new layer in the dockerfile, and there are typically only 128 layers allowed (apparently it is actually an undocumented amount: https://github.com/docker/docker.github.io/issues/8230).

It is gross, but the "best practice" is to chain all of these together into a single `RUN` to save layers.

Reply:
I really don't have much experience with docker, so very much appreciate this info!  Will change.

Reply:
BTW, this reminds me of [_"INTERCAL, YAML, And Other Horrible Programming Languages"_](https://blog.earthly.dev/intercal-yaml-and-other-horrible-programming-languages/)...